app.py

```diff
@@ -207,10 +240,152 @@ def plot_data(node_id):
     return pn.Column(node_id_input, dynamic_plot, tabulator)
 
 
+def page5():
+    hv.extension("matplotlib")
+    num_sites = df_trees.num_sites
+    max_num_sites = int(np.max(num_sites))
+    min_num_sites = int(np.min(num_sites))
+    sites_default_num_bins = int(np.sqrt(max_num_sites))
+
+    sites_log_checkbox = pn.widgets.Checkbox(
+        name="log y-axis", value=False, align=("center")
+    )
+
+    sites_xlim_slider = pn.widgets.IntRangeSlider(
+        name="x limits",
+        start=min_num_sites,
+        value=(min_num_sites, max_num_sites),
+        end=max_num_sites,
+        step=10,  # TODO set a sensible step size
+        align=("center"),
+    )
+
+    sites_bins_slider = pn.widgets.IntSlider(
+        name="Number of bins",
+        start=1,
+        end=min(200, int(max_num_sites / 2)),  # TODO set a sensible end value
+        value=min(200, int(max_num_sites / 2), sites_default_num_bins),
+        step=1,  # TODO set a sensible step size
+        align=("center"),
+    )
+
+    sites_reset_checkbox = pn.widgets.Checkbox(
+        name="reset", value=False, align=("center")
+    )
+
+    def reset_sites(reset=False):
+        if reset:
+            sites_log_checkbox.value = False
+            sites_xlim_slider.value = (min_num_sites, max_num_sites)
+            sites_bins_slider.value = sites_default_num_bins
+            sites_reset_checkbox.value = False
+
+    sites_hist_func = make_hist_func(
+        num_sites,
+        "Sites per tree",
+        num_bins=sites_default_num_bins,
+        log_y=True,
+        xlim=(min_num_sites, max_num_sites),
+    )
```

Comment:
```suggestion
    sites_hist_func = functools.partial(
        make_hist_matplotlib,
        num_sites,
        "Sites per tree",
        bins_range=sites_default_num_bins,
        log_y=True,
        xlim=(min_num_sites, max_num_sites),
    )
```

You don't need the ``make_hist_func`` function then